### PR TITLE
Allow configuring of tolerations for ingress-nginx daemonset

### LINKED ISSUE #720: Add tolerations to ingress-nginx
Essentially we need a way to add tolerations via cluter.yml to ingress-nginx daemonset to allow it deployed on tainted hosts too

### PR DESCRIPTION
**Pull request by jnummelin**
fixes #720 

Tolerations can be configured using something like:
```
addons:
  ingress-nginx:
    enabled: true
    tolerations:
      - key: "key"
        operator: "Equal"
        value: "value"
        effect: "NoSchedule"
```

### CODE REVIEW

File: addons/ingress-nginx/resources/daemonset.yml.erb [hunk 1]
```diff
@@ -25,6 +25,13 @@ spec:
       <%- end -%>
       serviceAccountName: nginx-ingress-serviceaccount
       hostNetwork: true
+      tolerations:
```
Comment (jakolehm) [inline addons/ingress-nginx/resources/daemonset.yml.erb]: Probably should not render `tolerations:` at all if not set?

Comment (jnummelin) [reply addons/ingress-nginx/resources/daemonset.yml.erb]: The end result is the same, there's no tolerations. :) But I can guard that too if needed

Comment (jakolehm) [reply addons/ingress-nginx/resources/daemonset.yml.erb]: I think it looks better if guarded.

Comment (jnummelin) [reply addons/ingress-nginx/resources/daemonset.yml.erb]: done

**Pull request merged**